File: Leetcode/cpp/sol867.cpp
//https://leetcode.com/problems/transpose-matrix/description/

class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        vector <vector<int>> result;
        for(auto col = 0; col <matrix[0].size(); col++){
            vector<int> temp = {};
            for(auto row = 0; row<matrix.size(); row++){
                temp.push_back(matrix[row][col]);
            }
            result.push_back(temp);
        }

        return result;

    }
};

File: Leetcode/python/sol867.py
# https://leetcode.com/problems/transpose-matrix/
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rslt = []
        for i in range(len(matrix[0])):
            temp = [matrix[j][i] for j in range(len(matrix))]
            rslt.append(temp)
        return rslt

